### DISCUSSION TITLE
How to batch Deskew multiple image files

### DISCUSSION THREAD

**Discussion by dmcnaugh15** [ImageMagick/ImageMagick#7850] [Help] [answered]
### ImageMagick version

7.1.1-41-Q16-HDRI-x64-dll

### Operating system, version and so on

Windows 11

### Description

Disclaimer: Super noob status

Help request: I want to batch deskew up to 300 images in a specific folder.  I know how to deskew 1 image at a time.  I have no idea how to do it for multiple images.  There is this post on a similar topic (https://github.com/ImageMagick/ImageMagick/discussions/2511), but I couldn't figure out a solution from that.  Can anyone point me in the right direction?

**Comment by fmw42** [comment, marked as answer]
Use magick mogrify.  It will automatically loop through a given folder (but will no go into subfolders) and process each image in the folder. The output folder must already exist.

```
cd some_folder (as the current folder)
magick mogrify -path path_to_output_folder -format format_for_output -deskew 40% * (or *.suffix)

```
Where * gets every file in the current_folder

See https://usage.imagemagick.org/basics/#mogrify

**Comment by dmcnaugh15** [comment]
@fmw42 Perfect.  Worked for me.  Thank you!
